src/components/Shared/Filters/FilterPanel.tsx

```diff
@@ -460,91 +440,48 @@ export const FilterPanel: React.FC<FilterPanelProps & BoxProps> = ({
                 }
                 return {
                   ...acc,
-                  [key]: value.split(',').map((enumValue) => {
-                    // --any--,none
-                    switch (enumValue) {
-                      case 'Appointment':
-                        return ActivityTypeEnum.Appointment;
-                      case 'Call':
-                        return ActivityTypeEnum.Call;
-                      case 'Email':
-                        return ActivityTypeEnum.Email;
-                      case 'Facebook Message':
-                        return ActivityTypeEnum.FacebookMessage;
-                      case 'Prayer Request':
-                        return ActivityTypeEnum.PrayerRequest;
-                      case 'Talk to In Person':
-                        return ActivityTypeEnum.TalkToInPerson;
-                      case 'Text Message':
-                        return ActivityTypeEnum.TextMessage;
-                      case 'Thank':
-                        return ActivityTypeEnum.Thank;
-                      case 'None':
-                        return ActivityTypeEnum.None;
-                      case 'Letter':
-                        return ActivityTypeEnum.Letter;
-                      case 'Newsletter - Physical':
-                        return ActivityTypeEnum.NewsletterPhysical;
-                      case 'Newsletter - Email':
-                        return ActivityTypeEnum.NewsletterEmail;
-                      case 'Pre Call Letter':
-                        return ActivityTypeEnum.PreCallLetter;
-                      case 'Reminder Letter':
-                        return ActivityTypeEnum.ReminderLetter;
-                      case 'Support Letter':
-                        return ActivityTypeEnum.SupportLetter;
-                      case 'To Do':
-                        return ActivityTypeEnum.ToDo;
-                      default:
-                        return ActivityTypeEnum.None;
-                    }
-                  }),
+                  [key]: value
+                    .split(',')
+                    .map((enumValue) => {
+                      // --any--,none
+                      return (
+                        (activities?.find((activity) => {
+                          return (
+                            activity.id === enumValue ||
+                            activity.value === enumValue ||
+                            activity?.id?.toLowerCase() === enumValue
+                          );
+                        })?.id as ActivityTypeEnum) || ActivityTypeEnum.None
+                      );
+                    })
+                    .flat(),
```

Comment:
Why are we flattening? I thought we're working with an array of `ActivityTypeEnum` not, an array of arrays.